src/hooks/useChannelSwitch.jsx

```diff
@@ -1,34 +1,44 @@
 import { useCallback } from "react";
 
 import { getRandomNoAdChannel } from "@/apis/radioChannels";
+import { SETTING_TYPES } from "@/constants/settingOptions";
 import { useChannelStore } from "@/store/useChannelStore";
+import { useUserStore } from "@/store/useUserStore";
+import { controlStreamingSwitch } from "@/utils/playControl";
 
-const useChannelSwitch = () => {
+const useChannelSwitch = (videoRef) => {
   const prevChannelId = useChannelStore((state) => state.prevChannelId);
   const setPrevChannelId = useChannelStore((state) => state.setPrevChannelId);
+  const { settings } = useUserStore();
+  const isAdDetect = settings[SETTING_TYPES.AD_DETECT];
 
   const handleChannelSwitch = useCallback(
     async (isAd) => {
       try {
         let channelId = 0;
         if (isAd) {
           try {
-            channelId = await getRandomNoAdChannel();
+            const { data } = await getRandomNoAdChannel();
+            channelId = data.id;
             setPrevChannelId(channelId);
           } catch (error) {
             console.error("fetch randomNoAdchannel failed", error);
           }
         } else {
+          if (prevChannelId == null) {
```

Comment:
`null`인지 확인하고 싶은거라면 `===`를 사용해도 좋을 것 같습니다!

Reply:
좋은 리뷰 감사해요!
prevChannelId가 0인 경우가 있을 수 있어 `!prevChannelId`를 사용하지 않았고
null과 undefined 둘 다 확인하고 싶어 느슨한 동등 비교(`==`)를 사용했습니다. ☺️

Reply:
저는 희주님께서 말씀해주신 대로
```suggestion
          if (prevChannelId === null || prevChannelId === undefine) {
```
이게 더 좋을 것 같습니다!

Reply:
`prevChannelId`가 0일 수 있다는 전제라면 `!prevChannelId`보다는 `prevChannelId === null || prevChannelId === undefined`이 의도를 더 명확하게 보여줘서 저도 이쪽이 더 자연스러운 것 같아요‼️

Reply:
좋은 의견 감사합니다!
의도를 보다 명확하게 전달할 수 있도록 prevChannelId === null || prevChannelId === undefined 형태로 수정하겠습니다 🙂